vertx-web-client/src/main/java/io/vertx/ext/web/client/cache/CacheAdapter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.web.client.cache;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.client.impl.cache.CacheKey;
+import io.vertx.ext.web.client.impl.cache.CachedHttpResponse;
+
+/**
+ * An API to store and retrieve HTTP responses.
+ *
+ * @author <a href="mailto:craigday3@gmail.com">Craig Day</a>
+ */
+public interface CacheAdapter {
```

Comment:
it should be renamed to simply Cache or WebCache

Reply:
I renamed it to `CacheStore` to be consistent with `CookieStore`. `Cache` feels overly generic, but I can rename it to that if you would prefer.